### DISCUSSION TITLE
is it possible in a route guard to tell if a navigation was generated programmatically versus by a URL change in browser?

### DISCUSSION THREAD

**Discussion by gureckis** [vuejs/router#2258] [Help and Questions] [answered]
I want to be able to tell the difference between when my code calls `router.go(1)` (for instance) and if the user initiates a route change in the brower by typing the URL into the page.  is there some signal about that in Vue Router?

**Comment by posva** [comment, marked as answer]
You can wrap `router.go()` but there is no native way to tell where the navigation comes from. It will be possible to know with [the navigation api](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API) once it becomes widely available
